Zend/zend_compile.c

```diff
@@ -4712,6 +4712,125 @@ static void zend_compile_ns_call(znode *result, znode *name_node, zend_ast *args
 }
 /* }}} */
 
+static zend_op *zend_compile_rope_add(znode *result, uint32_t num, znode *elem_node);
+static zend_op *zend_compile_rope_add_ex(zend_op *opline, znode *result, uint32_t num, znode *elem_node);
+static void zend_compile_rope_finalize(znode *result, uint32_t j, zend_op *init_opline, zend_op *opline);
+
+static zend_result zend_compile_func_sprintf(znode *result, zend_ast_list *args) /* {{{ */
+{
+	znode format_string;
+
+	if (args->children < 1) {
+		return FAILURE;
+	}
+
+	zend_compile_expr(&format_string, args->child[0]);
+	if (!(format_string.op_type == IS_CONST && Z_TYPE(format_string.u.constant) == IS_STRING)) {
+		zend_do_free(&format_string);
+		return FAILURE;
+	}
+	if (!(Z_STRLEN(format_string.u.constant) < 32)) {
```

Comment:
Same here, why not just:

```suggestion
	if (Z_STRLEN(format_string.u.constant) >= 32) {
```

I also think this is too limiting. I think, especially for interpolated strings, length >=32 will exclude probably 95% of all cases.

Reply:
Increased the limit to 256 characters. I generally expect `sprintf()` format strings to be fairly short-ish (mine are generally shorter than 32 characters) and I don't want to parse huge format strings during compile time.

The limit can easily be adjusted later, once we have some real world data.

Reply:
@TimWolla If performance is a concern, you might be better off using `memchr` to find the next occurrence of `%`.

Reply:
That's good idea in any case, done. Found a mistake in handling `%%` that way.